articles/api-auth/tutorials/silent-authentication.md

```diff
@@ -1,45 +1,90 @@
+* An error response if the user is not logged in via SSO and therefore cannot be silently authenticated
+
+### Successful authentication response
+
+If the user was already logged in via SSO, Auth0 will respond exactly as if the user had authenticated manually through the SSO login page.
+For example, when using the [Authorization Code Grant](/api-auth/grant/authorization-code) (`response_type=code`, used for regular web applications), Auth0 would respond with an authorization code that can be exchanged for an ID token and optionally an access token:
+
+```
+GET https://your_callback_url/
+    ?code=...&
     state=...&
-    expires_in=3600
+    expires_in=...
+```
+
+Note that this response is indistinguishable from a login performed directly without the `prompt=none` parameter.
+
+### Error response
+
+If the user was not logged in via SSO or their SSO session had expired, Auth0 will redirect to the specified `redirect_uri` (callback URL) with an error:
+
 ```
+GET https://your_callback_url/
+    ?error=ERROR_CODE&
+    error_description=ERROR_DESCRIPTION&
+    state=...
+```
+
+When using the [Authorization Code Grant](/api-auth/grant/authorization-code), the error response parameters are returned in the query string.
+When using the [Implicit Grant](/api-auth/grant/implicit), they are returned in the hash fragment instead.
+
+The possible values for `ERROR_CODE` are defined by the [OpenID Connect specification](https://openid.net/specs/openid-connect-core-1_0.html#AuthError):
 
-This parameter indicates how many seconds the access token will be valid for.
-With this information we can call `setTimeout` or some other scheduling mechanism to renew the access token when needed.
+* `login_required`: The user was not logged in at Auth0, so silent authentication is not possible
+* `consent_required`: The user was logged in at Auth0, but needs to give consent to authorize the client
+* `interaction_required`: The user was logged in at Auth0 and has authorized the client, but needs to be redirected elsewhere before authentication can be completed; for example, when using a [redirect rule](/rules/redirect).
 
-## Use `prompt=none` to retrieve a new token
+If any of these errors are returned, the user must be redirected to the Auth0 login page without the `prompt=none` parameter to authenticate.
 
-When you authenticate a user to your application, you would normally redirect them to the [`/authorize` endpoint of Auth0's authentication API](/api/authentication#social).
+## Refreshing expired tokens
+
+Access tokens are opaque to clients. This means that clients are unable to inspect the contents of access tokens to determine their expiration date.
+
+There are two options to determine when an access token expires:
+
+1. Read the `expires_in` response parameter returned by Auth0
+2. Ignore expiration dates altogether. Instead, try to renew the access token if rejected by the API.
```

Comment:
I would rephrase the end of this sentence from:

> if rejected by the API.

to: 

> if your API rejects a request from the client (eg. with a `401`).

Just so its clear which API we're talking about (i.e. not Auth0's API).